src/udev.rs

```diff
@@ -44,6 +44,43 @@ pub fn get_udev_info(
     Ok(())
 }
 
+/// Lookup a udev attribute given the `port_path` and `attribute`.
+///
+/// This only works on Linux and not all devices have all attributes.
+/// These attributes are generally readable by all users.
+///
+/// ```no_run
+/// use cyme::udev::get_udev_attribute;
+///
+/// let interface_class = get_udev_attribute(&String::from("1-0:1.0"),"bInterfaceClass").unwrap();
+/// assert_eq!(interface_class, "09");
+/// ```
+pub fn get_udev_attribute<T: AsRef<std::ffi::OsStr> + std::fmt::Display>(
+    port_path: &String,
+    attribute: T,
+) -> Option<String> {
+    let path: String = format!("/sys/bus/usb/devices/{}", port_path);
+    let device = match udevlib::Device::from_syspath(&Path::new(&path)).map_err(|e| {
+        Error::new(
+            ErrorKind::Udev,
+            &format!(
+                "Failed to get udev attribute {} for device at {}: Error({})",
+                attribute,
+                path,
+                e.to_string()
+            ),
+        )
+    }) {
+        Ok(d) => d,
+        Err(err) => {
+            log::warn!("{:?}", err);
+            return None;
+        }
+    };
+    let value = device.attribute_value(attribute).unwrap_or_default();
```

Comment:
This should  return `device.attribute_value(attribute).map(|s| s.to_str().unwrap_or("").to_string())` so that it will actually fall through to usb_ids if attribute_value returns `None`.